finagle-http/src/main/scala/com/twitter/finagle/http/filter/DtabFilter.scala

```diff
@@ -40,7 +40,41 @@ object DtabFilter {
     Future.value(rsp)
   }
 
+  @deprecated("use DtabFilter.Decoder", "2016-06-07")
   class Finagle[Req <: Message] extends DtabFilter[Req, Response] {
     def respondToInvalid(req: Req, msg: String) = invalidResponse(msg)
   }
+
+  /**
+   * Extracts Dtab-local headers from incoming requests and adds the
+   * Dtab to the local context.
+   */
+  class Extractor extends DtabFilter[Request, Response] {
+    def respondToInvalid(req: Request, msg: String) = invalidResponse(msg)
+  }
+
+  /**
+   * Modifies each request with Dtab encoding from Dtab.local and
+   * streams chunked responses via `Reader`.  If the request already
+   * contains Dtab headers they will be dropped silently.
+   */
+  class Injector extends SimpleFilter[Request, Response] {
+    def apply(req: Request, service: Service[Request, Response]): Future[Response] = {
+      // XXX won't this behave poorly with retries?
+      // val dtabHeaders = HttpDtab.strip(req)
+      // if (dtabHeaders.nonEmpty) {
+      //   // Log an error immediately if we find any Dtab headers already in the request and report them
+      //   val headersString = dtabHeaders.map({case (k, v) => s"[$k: $v]"}).mkString(", ")
+      //   log.error(s"discarding manually set dtab headers in request: $headersString\n" +
+      //     s"set Dtab.local instead to send Dtab information.")
+      // }
```

Comment:
I just commented out for the purposes of discussion. As I mentioned earlier, I think that this error reporting should be removed, since it does not play well with retries.